File: frontend/lib/models/category.dart
class Category {
  final int id;
  final String name;
  final String nameRu;
  final String slug;
  final bool isActive;
  final int displayOrder;

  const Category({
    required this.id,
    required this.name,
    required this.nameRu,
    required this.slug,
    required this.isActive,
    required this.displayOrder,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'] ?? '',
      nameRu: json['name_ru'] ?? json['name'] ?? '',
      slug: json['slug'] ?? '',
      isActive: json['is_active'] ?? true,
      displayOrder: json['display_order'] ?? 0,
    );
  }
}

File: backend/models/category.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from database import Base


class Category(Base):
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100), nullable=False)
    name_ru = Column(String(100), nullable=True)
    slug = Column(String(100), nullable=False, unique=True, index=True)
    parent_id = Column(Integer, ForeignKey("categories.id", ondelete="SET NULL"), nullable=True, index=True)
    is_active = Column(Boolean, default=True, index=True)
    display_order = Column(Integer, default=0)
    created_at = Column(DateTime, server_default=func.now())

    # Relationships
    parent = relationship("Category", remote_side="Category.id", backref="subcategories")
    listings = relationship("Listing", back_populates="category")
